source/uwp/AdaptiveCardsObjectModel/AdaptiveCardsObjectModel.vcxproj

```diff
@@ -126,100 +124,100 @@
-    <ClInclude Include="lib\AdaptiveShowCardActionParser.h" />
-    <ClInclude Include="lib\AdaptiveSubmitActionParser.h" />
-    <ClInclude Include="lib\AdaptiveTable.h" />
-    <ClInclude Include="lib\AdaptiveTableCell.h" />
-    <ClInclude Include="lib\AdaptiveTableColumnDefinition.h" />
-    <ClInclude Include="lib\AdaptiveTableParser.h" />
-    <ClInclude Include="lib\AdaptiveTableRow.h" />
-    <ClInclude Include="lib\AdaptiveTextBlockParser.h" />
-    <ClInclude Include="lib\AdaptiveTextElement.h" />
-    <ClInclude Include="lib\AdaptiveTextInputParser.h" />
-    <ClInclude Include="lib\AdaptiveTextRun.h" />
-    <ClInclude Include="lib\AdaptiveTimeInputParser.h" />
-    <ClInclude Include="lib\AdaptiveToggleInputParser.h" />
-    <ClInclude Include="lib\AdaptiveToggleVisibilityActionParser.h" />
-    <ClInclude Include="lib\AdaptiveToggleVisibilityTarget.h" />
-    <ClInclude Include="lib\AdaptiveToggleVisibilityAction.h" />
-    <ClInclude Include="lib\AdaptiveTokenExchangeResource.h" />
-    <ClInclude Include="lib\AdaptiveUnsupportedAction.h" />
-    <ClInclude Include="lib\AdaptiveUnsupportedElement.h" />
-    <ClInclude Include="lib\AdaptiveWarning.h" />
-    <ClInclude Include="lib\CustomActionWrapper.h" />
-    <ClInclude Include="lib\CustomElementWrapper.h" />
-    <ClInclude Include="lib\AdaptiveCard.h" />
-    <ClInclude Include="lib\AdaptiveChoiceInput.h" />
-    <ClInclude Include="lib\AdaptiveChoiceSetInput.h" />
-    <ClInclude Include="lib\AdaptiveColumn.h" />
-    <ClInclude Include="lib\AdaptiveColumnSet.h" />
-    <ClInclude Include="lib\AdaptiveContainer.h" />
-    <ClInclude Include="lib\AdaptiveFact.h" />
-    <ClInclude Include="lib\AdaptiveFactSet.h" />
-    <ClInclude Include="lib\AdaptiveImage.h" />
-    <ClInclude Include="lib\AdaptiveImageSet.h" />
-    <ClInclude Include="lib\AdaptiveDateInput.h" />
-    <ClInclude Include="lib\AdaptiveNumberInput.h" />
-    <ClInclude Include="lib\AdaptiveTextInput.h" />
-    <ClInclude Include="lib\AdaptiveTimeInput.h" />
-    <ClInclude Include="lib\AdaptiveToggleInput.h" />
-    <ClInclude Include="lib\AdaptiveOpenUrlAction.h" />
-    <ClInclude Include="lib\AdaptiveShowCardAction.h" />
-    <ClInclude Include="lib\AdaptiveSubmitAction.h" />
-    <ClInclude Include="lib\AdaptiveTextBlock.h" />
-    <ClInclude Include="lib\pch.h" />
-    <ClInclude Include="lib\ObjectModelUtil.h" />
-    <ClCompile Include="lib\AdaptiveBackgroundImage.cpp" />
-    <ClCompile Include="lib\AdaptiveMedia.cpp">
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveAuthCardButton.cpp" />
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveAuthentication.cpp" />
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveCaptionSource.cpp" />
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveContentSource.cpp" />
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveExecuteAction.cpp" />
+    <ClCompile Include="..\sharedObjectModel\lib\AdaptiveExecuteActionParser.cpp" />
```

Comment:
I wonder if this wouldn't be a good candidate for using a "Shared Items Project" -- this way as files get added to the object model, they only need to be added to the shared items project, which will automatically get picked up by project reference elsewhere.

[reference](https://learn.microsoft.com/en-us/visualstudio/ide/managing-references-in-a-project?view=vs-2022#shared-project-references)

Reply:
Great idea! This is much cleaner.

Reply:
This ended up being an easy piece to break out, both making this PR smaller, and solving the lost file history problem I mentioned under jwoo-msft's comment above. New PR, intended as a prerequisite to this one: https://github.com/microsoft/AdaptiveCards/pull/8203